# Analysis  on IMU (Inertial Measurement Unit) Dataset using python

## Introduction To Dataset

An inertial measurement unit (IMU) measures and reports raw or filtered angular rate and specific force/acceleration experienced by the object it is attached to. Data outputs for an IMU are typically body-frame accelerations,
angular rates, and (optionally) magnetic field measurements.

Variables used in this dataset is

1. Category = The category to which this IMU belongs.


2. Manufacturer = Name of the manufacturer.


3. part number = Discrete part number


4. sku = stock keeping unit


5. alt_sku = alternate stock keeping unit


6. Description = description of the product


7. Status = Whether it is in stock or not


8. Quantity = Quantity available


9. Price = Price of the product


10. sku_create_date: when was the product procured


## Understanding of dataset

In this dataset are variables are self explanatory

## Important necessary libraries

In [1]:
import pandas as pd   # for analysis
import numpy as np     # for numerical calculation                  

## Reading imu dataset

In [2]:
IMU_data = pd.read_excel("imus+(2).xlsx")   # read excel file using pandas
IMU_data

category                manufacturer      part_number     sku  \
0            1539  MS-OPEN VALUE SUBSCRIPTION        021-09108  BA0011   
1            1539  MS-OPEN VALUE SUBSCRIPTION        059-07014  BA0025   
2            1539  MS-OPEN VALUE SUBSCRIPTION        065-06444  BA0039   
3            1539  MS-OPEN VALUE SUBSCRIPTION        076-04355  BA0067   
4            1582  MS-OPEN VALUE SUBSCRIPTION        077-05318  BA0080   
...           ...                         ...              ...     ...   
1048570      1241          IOSAFE DIRECT SHIP  75200-3838-1500  6QP586   
1048571      1241          IOSAFE DIRECT SHIP  75200-3838-1500  6QP586   
1048572      1241          IOSAFE DIRECT SHIP  75200-3840-1500  6QP587   
1048573      1241          IOSAFE DIRECT SHIP  75200-3840-1500  6QP587   
1048574      1241          IOSAFE DIRECT SHIP  75200-3840-1500  6QP587   

        alt_sku                                        description    status  \
0           NaN               1YR OFFICESTD ALNG LICSAPK OLV, D AP  No Stock   
1           NaN                    1YR WORD ALNG LICSAPK OLV D AP,  No Stock   
2           NaN                   1YR EXCEL ALNG LICSAPK OLV D AP,  No Stock   
3           NaN                   1YR PRJCT ALNG LICSAPK OLV D AP,  No Stock   
4           NaN                  1YR ACCESS ALNG LICSAPK OLV D, AP  No Stock   
...         ...                                                ...       ...   
1048570     NaN  IOSAFE 1019+ NAS 5X4TB WITH 5YR, DATA RECOVERY...  No Stock   
1048571     NaN  IOSAFE 1019+ NAS 5X4TB WITH 5YR, DATA RECOVERY...  No Stock   
1048572     NaN  IOSAFE 1019+ NAS 5X8TB WITH 5YR, DATA RECOVERY...  No Stock   
1048573     NaN  IOSAFE 1019+ NAS 5X8TB WITH 5YR, DATA RECOVERY...  No Stock   
1048574     NaN  IOSAFE 1019+ NAS 5X8TB WITH 5YR, DATA RECOVERY...  No Stock   

         qty    price sku_create_date  
0          0   143.58      2020-08-17  
1          0    56.09      2020-08-17  
2          0    56.09      2020-08-17  
3          0   217.60      2020-08-17  
4          0    56.09      2020-08-17  
...      ...      ...             ...  
1048570    0  4557.08      2020-08-17  
1048571    0  4557.08      2020-08-17  
1048572    0  7034.83      2020-08-17  
1048573    0  6836.14      2020-08-17  
1048574    0  6836.14      2020-08-17  

[1048575 rows x 10 columns]

In [3]:
df=pd.concat(pd.read_excel("imus+(2).xlsx",sheet_name=None),ignore_index=True)  # concate sheets in excel file
df

category                manufacturer part_number     sku alt_sku  \
0            1539  MS-OPEN VALUE SUBSCRIPTION   021-09108  BA0011     NaN   
1            1539  MS-OPEN VALUE SUBSCRIPTION   059-07014  BA0025     NaN   
2            1539  MS-OPEN VALUE SUBSCRIPTION   065-06444  BA0039     NaN   
3            1539  MS-OPEN VALUE SUBSCRIPTION   076-04355  BA0067     NaN   
4            1582  MS-OPEN VALUE SUBSCRIPTION   077-05318  BA0080     NaN   
...           ...                         ...         ...     ...     ...   
2627921      9945        VERBATIM CORPORATION       93975  986034     NaN   
2627922      9945        VERBATIM CORPORATION       93975  986034     NaN   
2627923      9945        VERBATIM CORPORATION       93975  986034     NaN   
2627924      9945        VERBATIM CORPORATION       93975  986034     NaN   
2627925      9945        VERBATIM CORPORATION       93975  986034     NaN   

                                               description    status  qty  \
0                     1YR OFFICESTD ALNG LICSAPK OLV, D AP  No Stock    0   
1                          1YR WORD ALNG LICSAPK OLV D AP,  No Stock    0   
2                         1YR EXCEL ALNG LICSAPK OLV D AP,  No Stock    0   
3                         1YR PRJCT ALNG LICSAPK OLV D AP,  No Stock    0   
4                        1YR ACCESS ALNG LICSAPK OLV D, AP  No Stock    0   
...                                                    ...       ...  ...   
2627921  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   12   
2627922  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   12   
2627923  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   12   
2627924  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   12   
2627925  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   12   

          price sku_create_date  
0        143.58      2020-08-17  
1         56.09      2020-08-17  
2         56.09      2020-08-17  
3        217.60      2020-08-17  
4         56.09      2020-08-17  
...         ...             ...  
2627921   50.53      2020-08-28  
2627922   50.53      2020-08-28  
2627923   50.53      2020-08-28  
2627924   50.53      2020-08-28  
2627925   50.53      2020-08-28  

[2627926 rows x 10 columns]

shape of dataset after concate the dataset

In [6]:
df.shape    # no of rows and columns in data

(2627926, 10)

All details about Imu dataset

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2627926 entries, 0 to 2627925
Data columns (total 10 columns):
 #   Column           Dtype         
---  ------           -----         
 0   category         int64         
 1   manufacturer     object        
 2   part_number      object        
 3   sku              object        
 4   alt_sku          object        
 5   description      object        
 6   status           object        
 7   qty              int64         
 8   price            float64       
 9   sku_create_date  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(2), object(6)
memory usage: 200.5+ MB


## In Stock Products

In [10]:
In_stock = df[df.status == 'In Stock']   #Separate in stock products data

In [11]:
In_stock

category                    manufacturer         part_number     sku  \
253          9920  CASE LOGIC-PERSONAL & PORTABLE             3200943  BA6371   
271           531                OMNITRON SYSTEMS              7206-0  BA6834   
320          9070         ZEBRA ENTERPRISE MCD-A1       SAC5070-800CR  BC0227   
324          9080         ZEBRA ENTERPRISE MCD-A1  KT-CLMPT-RS507-01R  BC0232   
326          9080         ZEBRA ENTERPRISE MCD-A1    KT-PAD-RS507-10R  BC0234   
...           ...                             ...                 ...     ...   
2627921      9945            VERBATIM CORPORATION               93975  986034   
2627922      9945            VERBATIM CORPORATION               93975  986034   
2627923      9945            VERBATIM CORPORATION               93975  986034   
2627924      9945            VERBATIM CORPORATION               93975  986034   
2627925      9945            VERBATIM CORPORATION               93975  986034   

        alt_sku                                        description    status  \
253         NaN  LAPTOP ROLLING CASE BLACK, SECURITY FRIENDLY 17IN  In Stock   
271         NaN  1000BASE-SX SFP LC/MM/DF 850NM, 550M PLUGGABLE...  In Stock   
320         NaN  RS507 8SLOT BATTERY CHARGE KIT, EXT PWR SUPL/U...  In Stock   
324         NaN  RS507 TRIGGER SWIVEL DOES NOT, INCLUDE STRAP/B...  In Stock   
326         NaN                      10 CT COMFORT PADS FOR RS507,  In Stock   
...         ...                                                ...       ...   
2627921     NaN  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   
2627922     NaN  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   
2627923     NaN  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   
2627924     NaN  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   
2627925     NaN  CD & DVD TRIMPAK CASES 200PK, CLEAR FLEXIBLE P...  In Stock   

         qty   price sku_create_date  
253        7   99.00      2020-08-17  
271        2   59.09      2020-08-17  
320        3  500.06      2020-08-17  
324      115   86.34      2020-08-17  
326       42   51.80      2020-08-17  
...      ...     ...             ...  
2627921   12   50.53      2020-08-28  
2627922   12   50.53      2020-08-28  
2627923   12   50.53      2020-08-28  
2627924   12   50.53      2020-08-28  
2627925   12   50.53      2020-08-28  

[713788 rows x 10 columns]

### Q1. Which are the top Manufacturers with price, quantity

In [13]:
In_stock['netprice'] = In_stock.qty*In_stock.price      # get netprice by price and qty product

C:\Users\ashis\AppData\Local\Temp/ipykernel_4828/2529896364.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  In_stock['netprice'] = In_stock.qty*In_stock.price


In [14]:
In_stock.sort_values("netprice", ascending=False).head(10) # use of sort function to get the top Manufacturers with price,quantity

category                 manufacturer part_number     sku alt_sku  \
2502762      1221          DELL ESG WARRANTIES    823-7912  7GZ792     NaN   
1890636      1221  DELL CSG SERVICE WARRANTIES    818-6447  3DB700     NaN   
625985       1221  DELL CSG SERVICE WARRANTIES    818-6447  3DB700     NaN   
2450786      1221  DELL CSG SERVICE WARRANTIES    818-6447  3DB700     NaN   
2502761      1221          DELL ESG WARRANTIES    823-7911  7GZ791     NaN   
2450785      1221          DELL ESG WARRANTIES    818-6446  3DB699     NaN   
1890635      1221          DELL ESG WARRANTIES    818-6446  3DB699     NaN   
625984       1221          DELL ESG WARRANTIES    818-6446  3DB699     NaN   
2450784      1221          DELL ESG WARRANTIES    818-6445  3DB698     NaN   
1664364      1251  IM CHOICE ADVANTAGE - GOVED     FEDADV1  ZM9366     NaN   

                                             description    status   qty  \
2502762                        3YR NBD TO 5YR PROSPT 4H,  In Stock  9999   
1890636                             5YR PROSUPPORT PLUS,  In Stock  9999   
625985                              5YR PROSUPPORT PLUS,  In Stock  9999   
2450786                             5YR PROSUPPORT PLUS,  In Stock  9999   
2502761                        3YR NBD TO 5YR PROSPT 4H,  In Stock  9999   
2450785                             5YR PROSUPPORT PLUS,  In Stock  9999   
1890635                             5YR PROSUPPORT PLUS,  In Stock  9999   
625984                              5YR PROSUPPORT PLUS,  In Stock  9999   
2450784                             5YR PROSUPPORT PLUS,  In Stock  9999   
1664364  FED ADVANTAGE CONTRACT SUPPORT-, STARTING LEVEL  In Stock  9999   

           price sku_create_date     netprice  
2502762  4236.30      2020-08-26  42358763.70  
1890636  3800.72      2020-08-24  38003399.28  
625985   3800.72      2020-08-17  38003399.28  
2450786  3798.68      2020-08-26  37983001.32  
2502761  3111.01      2020-08-26  31106988.99  
2450785  3019.19      2020-08-26  30188880.81  
1890635  2804.29      2020-08-24  28040095.71  
625984   2804.29      2020-08-17  28040095.71  
2450784  2600.54      2020-08-26  26002799.46  
1664364  2502.50      2020-08-24  25022497.50

Here, Top Manufacturers have  maximum price with the more  quantity remaining.

OR

In [26]:
top_man = In_stock.sort_values("price", ascending=False)

In [46]:
top_man

category         manufacturer       part_number     sku alt_sku  \
1199874       521      HPE - SWITCHING        JL376A#B2B  6RN996     NaN   
2571702       295     HPE - NIMBLE BTO            R3S72A  7JQ609     NaN   
2620934       295     HPE - NIMBLE BTO            R3S72A  7JQ609     NaN   
1203979       505  HPE - ARUBA INSTANT            JW830A  7U5737     NaN   
1203946       505  HPE - ARUBA INSTANT            JW675A  7U4379     NaN   
...           ...                  ...               ...     ...     ...   
1523294      2201                  C2G             01921  U40479     NaN   
1287203      1251             CONFIG 1    BTRY-MC55EAB00  CX4544     NaN   
21676        1251             CONFIG 1  SCX-5835FN - TAA  CX4513     NaN   
21677        1251             CONFIG 1    BTRY-MC55EAB00  CX4544     NaN   
1287202      1251             CONFIG 1  SCX-5835FN - TAA  CX4513     NaN   

                                               description    status  qty  \
1199874  ARUBA 8400 1 MM 3 PS 2 FM 32/8, BDL JMPCBL-NA/...  In Stock    3   
2571702  HPE NS AF20 ALL FLASH 23TB BDL, 1 OR 3 YR SUPP...  In Stock    2   
2620934  HPE NS AF20 ALL FLASH 23TB BDL, 1 OR 3 YR SUPP...  In Stock    3   
1203979         ARUBA 7240XM US FIPS/TAA, CONTROLLER PL=VL  In Stock    1   
1203946                         7240XMDC CONTROLLER, PL=VL  In Stock    3   
...                                                    ...       ...  ...   
1523294                  RJ11 4X4 MOD PLUG FLAT STRND, CBL  In Stock   21   
1287203  MC55 STANDARD SPARE BATTERY, W/ LABEL  SHEET  ...  In Stock   71   
21676                       SCX-5835FN MONO LASER P/S/C/F,  In Stock    1   
21677    MC55 STANDARD SPARE BATTERY, W/ LABEL  SHEET  ...  In Stock   71   
1287202                     SCX-5835FN MONO LASER P/S/C/F,  In Stock    1   

            price sku_create_date   netprice  
1199874  69729.03      2020-08-20  209187.09  
2571702  40893.99      2020-08-27   81787.98  
2620934  40893.99      2020-08-28  122681.97  
1203979  35554.13      2020-08-20   35554.13  
1203946  29385.55      2020-08-20   88156.65  
...           ...             ...        ...  
1523294      0.20      2020-08-24       4.20  
1287203      0.00      2020-08-24       0.00  
21676        0.00      2020-08-17       0.00  
21677        0.00      2020-08-17       0.00  
1287202      0.00      2020-08-24       0.00  

[713788 rows x 11 columns]

Here Top Manufacturers have  maximum price with the less  quantity remaining.

### Q2. Which are the most profitable products?

In [28]:
top_man = In_stock.sort_values("price", ascending=False, )    # sort values of price

In [32]:
top_man.loc[:,["part_number",'description'] ].head(20)  # get most profitable products

part_number  \
1199874            JL376A#B2B   
2571702                R3S72A   
2620934                R3S72A   
1203979                JW830A   
1203946                JW675A   
1146919                R3S71A   
1688357  PSA7000-10G-SFP-PLUS   
2550004  PSA7000-10G-SFP-PLUS   
2435893  PSA7000-10G-SFP-PLUS   
1127383  PSA7000-10G-SFP-PLUS   
1236885  PSA7000-10G-SFP-PLUS   
1259818                JW778A   
1017049          MSN3700-CS2F   
2281149          MSN3700-CS2F   
2617435          MSN3700-CS2F   
1254161          MSN3700-CS2F   
1903330         MSN2700-CS2FC   
638771          MSN2700-CS2FC   
1724513                JL366A   
841992             JL479A#B2B   

                                               description  
1199874  ARUBA 8400 1 MM 3 PS 2 FM 32/8, BDL JMPCBL-NA/...  
2571702  HPE NS AF20 ALL FLASH 23TB BDL, 1 OR 3 YR SUPP...  
2620934  HPE NS AF20 ALL FLASH 23TB BDL, 1 OR 3 YR SUPP...  
1203979         ARUBA 7240XM US FIPS/TAA, CONTROLLER PL=VL  
1203946                         7240XMDC CONTROLLER, PL=VL  
1146919  HPE NS HF20 ADAPTIVE 21TB BDL, 1OR 3 Y SUPP RE...  
1688357           APPL 7000 BASE SYSTEM FIBER, 64CORE PREM  
2550004           APPL 7000 BASE SYSTEM FIBER, 64CORE PREM  
2435893           APPL 7000 BASE SYSTEM FIBER, 64CORE PREM  
1127383           APPL 7000 BASE SYSTEM FIBER, 64CORE PREM  
1236885           APPL 7000 BASE SYSTEM FIBER, 64CORE PREM  
1259818                7205-K12-128 K12 128 AP BNDL, PL=VL  
1017049  SPECTRUM-2 1U SWITCH 32PORT, NPI ENGINEERING S...  
2281149  SPECTRUM-2 1U SWITCH 32PORT, NPI ENGINEERING S...  
2617435  SPECTRUM-2 1U SWITCH 32PORT, NPI ENGINEERING S...  
1254161  SPECTRUM-2 1U SWITCH 32PORT, NPI ENGINEERING S...  
1903330  SPECTRUM BASED 100GBE 1U OPEN, ETHERNET SWITCH...  
638771   SPECTRUM BASED 100GBE 1U OPEN, ETHERNET SWITCH...  
1724513      ARUBA 8400X 6P 40G/100G QSFP28, ADV MOD PL-NV  
841992   ARUBA 8320 48 10/6 40 X472 5 2, BDL JMPCBL-NA/...

 Above are the most profitable products

### Q3. Which are the products where you have seen a decrease in the price?

In [67]:
pro_dec = In_stock.groupby(['part_number','sku_create_date'])['price'].max() # use group by to get decrease in price
pro_dec= pd.DataFrame(pro_dec)    # make dataframe
pro_dec.sort_values('price',ascending=False)  # sort descending to get decrese in price

price
part_number      sku_create_date          
JL376A#B2B       2020-08-20       69729.03
R3S72A           2020-08-27       40893.99
                 2020-08-28       40893.99
JW830A           2020-08-20       35554.13
JW675A           2020-08-20       29385.55
...                                    ...
03810            2020-08-17           0.46
01921            2020-08-24           0.21
                 2020-08-17           0.21
SCX-5835FN - TAA 2020-08-17           0.00
                 2020-08-24           0.00

[111056 rows x 1 columns]

### Q4. Which are the product where you have seen an increase in the price?

In [43]:
In_stock.groupby(['part_number','sku_create_date'])['price'].min()
pro_inc= pd.DataFrame(pro_dec)
pro_dec.sort_values('price',ascending=True)

price
part_number      sku_create_date          
SCX-5835FN - TAA 2020-08-24           0.00
                 2020-08-17           0.00
01921            2020-08-24           0.21
                 2020-08-17           0.21
03820            2020-08-24           0.46
...                                    ...
JW675A           2020-08-20       29385.55
JW830A           2020-08-20       35554.13
R3S72A           2020-08-28       40893.99
                 2020-08-27       40893.99
JL376A#B2B       2020-08-20       69729.03

[111056 rows x 1 columns]

### Q5. Which is the top category by price and quantity?

In [ ]:
top_man = In_stock.sort_values("price", ascending=False)  # sort by price

Top category by price

In [51]:
In_stock[In_stock.price == In_stock.price.max()]     # use max function

category     manufacturer part_number     sku alt_sku  \
1199874       521  HPE - SWITCHING  JL376A#B2B  6RN996     NaN   

                                               description    status  qty  \
1199874  ARUBA 8400 1 MM 3 PS 2 FM 32/8, BDL JMPCBL-NA/...  In Stock    3   

            price sku_create_date   netprice  
1199874  69729.03      2020-08-20  209187.09

 521 is the top category in terms of price of products

Top category by quantity

In [53]:
In_stock[In_stock.qty == In_stock.qty.max()]      # use max function

category                 manufacturer part_number     sku alt_sku  \
2364017      1221  DELL CSG SERVICE WARRANTIES    808-3177  4A1885     NaN   

                                 description    status    qty  price  \
2364017  3YR NBD TO 3YR PROSUP NBD, OPTIPLEX  In Stock  60063  38.76   

        sku_create_date    netprice  
2364017      2020-08-25  2328041.88

 1221 is the top category in terms of quantity of products

### Q6. Which products are aging?

In [54]:
# for getting Products are aging  , first arranging In stack data in tems of descending order of quantity
Top_aging = In_stock.sort_values("qty", ascending =False).head(10)
Top_aging

category                 manufacturer part_number     sku alt_sku  \
2364017      1221  DELL CSG SERVICE WARRANTIES    808-3177  4A1885     NaN   
1135792      1221  DELL CSG SERVICE WARRANTIES    808-3123  4A1836     NaN   
2609950      1221  DELL CSG SERVICE WARRANTIES    808-3108  4A1821     NaN   
1245234      1221  DELL CSG SERVICE WARRANTIES    808-3105  4A1818     NaN   
2558454      1221  DELL CSG SERVICE WARRANTIES    808-3105  4A1818     NaN   
1190334      1221  DELL CSG SERVICE WARRANTIES    808-3105  4A1818     NaN   
2558456      1221  DELL CSG SERVICE WARRANTIES    808-3177  4A1885     NaN   
1245236      1221  DELL CSG SERVICE WARRANTIES    808-3129  4A1842     NaN   
2558455      1221  DELL CSG SERVICE WARRANTIES    808-3126  4A1839     NaN   
1245235      1221  DELL CSG SERVICE WARRANTIES    808-3126  4A1839     NaN   

                                               description    status    qty  \
2364017                3YR NBD TO 3YR PROSUP NBD, OPTIPLEX  In Stock  60063   
1135792  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  60018   
2609950  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  60014   
1245234  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  60005   
2558454  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  60004   
1190334  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  60002   
2558456                3YR NBD TO 3YR PROSUP NBD, OPTIPLEX  In Stock  60000   
1245236  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  59999   
2558455  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  59996   
1245235  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...  In Stock  59996   

          price sku_create_date     netprice  
2364017   38.76      2020-08-25   2328041.88  
1135792  129.96      2020-08-19   7799939.28  
2609950  136.42      2020-08-28   8187109.88  
1245234   84.24      2020-08-21   5054821.20  
2558454   84.06      2020-08-27   5043936.24  
1190334   84.24      2020-08-20   5054568.48  
2558456   38.68      2020-08-27   2320800.00  
1245236  267.10      2020-08-21  16025732.90  
2558455  188.33      2020-08-27  11299046.68  
1245235  188.74      2020-08-21  11323645.04

In [56]:
# Products are aging which are maximum in terms of quantity have low price

Aging_prod = Top_aging.loc[:,["part_number",'qty','description']]
Aging_prod

part_number    qty                                        description
2364017    808-3177  60063                3YR NBD TO 3YR PROSUP NBD, OPTIPLEX
1135792    808-3123  60018  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
2609950    808-3108  60014  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
1245234    808-3105  60005  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
2558454    808-3105  60004  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
1190334    808-3105  60002  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
2558456    808-3177  60000                3YR NBD TO 3YR PROSUP NBD, OPTIPLEX
1245236    808-3129  59999  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
2558455    808-3126  59996  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...
1245235    808-3126  59996  1YR HW SVC WITH IN-HOME/OS SVC, AFTER REM DIAG...

Here the products which have the more quantity can be considered as the as product aging.

Because having more product can be considered as they are not getting sold

# No Stock  Products

Seperate No stock data from imu data

In [57]:
No_stock = df[df.status == 'No Stock']   # separate the no-stock data 

### Q1. Which are the top Manufacturers with price, quantity?

In [60]:
No_stock.sort_values("price", ascending=False).head(10)      # sort the values according to price

category                manufacturer         part_number     sku  \
1046052      1231                      ALTNET   ALT-TPM-240000000  6QN385   
2310142      1231                      ALTNET   ALT-TPM-240000000  6QN385   
1046051      1231                      ALTNET    ALT-TPM-80000000  6QN384   
2310141      1231                      ALTNET    ALT-TPM-80000000  6QN384   
973610       1241        JUNIPER ROW SERVICES  SV5-SWA-JCE-1KGBPS  6JB451   
2237750      1241        JUNIPER ROW SERVICES  SV5-SWA-JCE-1KGBPS  6JB451   
973606       1241        JUNIPER ROW SERVICES  SV3-SWA-JCE-1KGBPS  6JB398   
2237746      1241        JUNIPER ROW SERVICES  SV3-SWA-JCE-1KGBPS  6JB398   
2261823      1241  JUNIPER ENTERPRISE ROUTING  QSFP-100G-LR4-1000  6KG376   
997704       1241  JUNIPER ENTERPRISE ROUTING  QSFP-100G-LR4-1000  6KG376   

        alt_sku                                        description    status  \
1046052     NaN  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock   
2310142     NaN  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock   
1046051     NaN  ALTNET TPM UP TO 80000000 USD, LIST MNT UPTO 8...  No Stock   
2310141     NaN  ALTNET TPM UP TO 80000000 USD, LIST MNT UPTO 8...  No Stock   
973610      NaN                  ROW 5YR SWA SUP FOR, JCE-1000GBPS  No Stock   
2237750     NaN                  ROW 5YR SWA SUP FOR, JCE-1000GBPS  No Stock   
973606      NaN                  ROW 3YR SWA SUP FOR, JCE-1000GBPS  No Stock   
2237746     NaN                  ROW 3YR SWA SUP FOR, JCE-1000GBPS  No Stock   
2261823     NaN               1000 UNIT BNDL OF, QSFP-100GBASE-LR4  No Stock   
997704      NaN               1000 UNIT BNDL OF, QSFP-100GBASE-LR4  No Stock   

         qty        price sku_create_date  
1046052    0  84974533.10      2020-08-17  
2310142    0  84974533.10      2020-08-24  
1046051    0  28324844.37      2020-08-17  
2310141    0  28324844.37      2020-08-24  
973610     0  10449000.00      2020-08-17  
2237750    0  10449000.00      2020-08-24  
973606     0   6966000.00      2020-08-17  
2237746    0   6966000.00      2020-08-24  
2261823    0   5600000.00      2020-08-24  
997704     0   5600000.00      2020-08-17

### Q2. Which are the most profitable products?

In [61]:
No_stock[No_stock.price == No_stock.price.max()]  # use max function

category manufacturer        part_number     sku alt_sku  \
1046052      1231       ALTNET  ALT-TPM-240000000  6QN385     NaN   
2310142      1231       ALTNET  ALT-TPM-240000000  6QN385     NaN   

                                               description    status  qty  \
1046052  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock    0   
2310142  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock    0   

              price sku_create_date  
1046052  84974533.1      2020-08-17  
2310142  84974533.1      2020-08-24

### Q3. Which are the products where you have seen a decrease in the price

In [62]:
prc_dec=No_stock.groupby(['part_number','sku_create_date'])['price'].max()  # use group by function
prc_dec=pd.DataFrame(prc_dec)  # make dataframe
prc_dec.sort_values(by='price',ascending=False)  # sort values 

price
part_number        sku_create_date             
ALT-TPM-240000000  2020-08-17       84974533.10
                   2020-08-24       84974533.10
ALT-TPM-80000000   2020-08-24       28324844.37
                   2020-08-17       28324844.37
SV5-SWA-JCE-1KGBPS 2020-08-24       10449000.00
...                                         ...
AM10621-SLV        2020-08-17              0.00
                   2020-08-24              0.00
AM10631-SLV        2020-08-17              0.00
AM20111-HT         2020-08-17              0.00
AM10121-BX         2020-08-24              0.00

[639405 rows x 1 columns]

### Q4. Which are the products where you have seen an increase in the price?

In [63]:
prc_inc = No_stock.groupby(['part_number','sku_create_date'])['price'].max()  # use group by 
prc_inc = pd.DataFrame(prc_dec)    # make dataframe
prc_inc.sort_values(by='price',ascending= True)    # sort values

price
part_number        sku_create_date             
AM20311-HT         2020-08-17              0.00
AM10611-SLV        2020-08-17              0.00
AM10521-SLV        2020-08-24              0.00
                   2020-08-17              0.00
AM10511-SLV        2020-08-24              0.00
...                                         ...
SV5-SWA-JCE-1KGBPS 2020-08-17       10449000.00
ALT-TPM-80000000   2020-08-17       28324844.37
                   2020-08-24       28324844.37
ALT-TPM-240000000  2020-08-24       84974533.10
                   2020-08-17       84974533.10

[639405 rows x 1 columns]

### Q5. Which is the top category by price and quantity?

In [64]:
# Top category by price and quantity  as there is No stock means zero quantity 
No_stock[No_stock.price == No_stock.price.max()]

category manufacturer        part_number     sku alt_sku  \
1046052      1231       ALTNET  ALT-TPM-240000000  6QN385     NaN   
2310142      1231       ALTNET  ALT-TPM-240000000  6QN385     NaN   

                                               description    status  qty  \
1046052  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock    0   
2310142  ALTNET TPM UP TO 240000000 USD, LIST MNT UPTO ...  No Stock    0   

              price sku_create_date  
1046052  84974533.1      2020-08-17  
2310142  84974533.1      2020-08-24

1231 is the top category in No-stock data by price and quantity